instrumentation/hibernate/hibernate-3.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/CriteriaInstrumentation.java

```diff
@@ -50,17 +50,12 @@ public void transform(TypeTransformer transformer) {
   public static class CriteriaMethodAdvice {
 
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void startMethod(
-        @Advice.This Criteria criteria,
-        @Advice.Origin("#m") String name,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelHibernateOperation") HibernateOperation hibernateOperation,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-
-      callDepth = CallDepth.forClass(HibernateOperation.class);
+    public static Object startMethod(
```

Comment:
same question here about using explicit return type

```suggestion
    public static CallDepth startMethod(
```

Reply:
I think this should work, because `CallDepth` is injected into the bootstrap classloader.

Reply:
While it *might* work now means that once we remove things from the bootstrap CL things will start to break, so as a general rule I think we should document that only two sets of types should be used in the advice method signature:
- types from the JDK that are already in the bootstrap CL
- types that are part of the instrumented code classpath

Reply:
I've added this to the documentation : https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11546/commits/547d422eeab1a37f9f88fe33df1bdb731aede34b

Reply:
thanks!